docs/troubleshooting.md

```diff
@@ -0,0 +1,249 @@
+time="2019-11-21T14:40:21Z" level=debug msg="checking for existing installplan" channel=beta id=eFa1R namespace=my-subscription-namespace pkg=prometheus source=my-catalogsource sub=my-subscription
+```
+
+From the logs it's clear that the Catalog operator believes that there are no `CatalogSources` available. This is due to the fact that `Subscriptions` cannot install operators provided by `CatalogSources` that are not in the same namespace unless the `CatalogSource` is created in the `olm namespace`. We can fix this issue by moving the `Subscription` to the `my-catalogsource-namespace namespace`.
+
+```sh
+$ cat my-updated-subscription.yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: my-subscription
+  namespace: my-catalogsource-namespace
+spec:
+  channel: beta
+  name: prometheus
+  source: my-catalogsource
+  sourceNamespace: my-catalogsource-namespace
+  installPlanApproval: Automatic
+```
+
+A `CSV` should now be created:
+```sh
+$ kubectl -n my-catalogsource-namespace get csvs
+NAME                        DISPLAY               VERSION   REPLACES                    PHASE
+prometheusoperator.0.32.0   Prometheus Operator   0.32.0    prometheusoperator.0.27.0   Failed
+```
+
+Great a `CSV` was created but it is in the failing state.
+
+## ClusterServiceVersion Troubleshooting
+
+Continuing from the scenario described in the [Subscription Troubleshooting section](#subscription-troubleshooting), your cluster now has a `CSV` in the failed phase. The `CSV` has been placed in the failed phase because OLM encountered an unrecoverable error when attempting to install the operator.
+
+Our first step should be to check the status of the `CSV`. The OLM operator will constantly update the Status with useful information regarding the state of the `CSV`. You can check the status of your `CSV` with the following command:
+
+```sh
+$ kubectl -n my-catalogsource-namespace get csv prometheusoperator.0.32.0 -o yaml | yq r - status
+certsLastUpdated: null
+certsRotateAt: null
+conditions:
+- lastTransitionTime: "2019-11-21T14:45:58Z"
+  lastUpdateTime: "2019-11-21T14:45:58Z"
+  message: csv in namespace with no operatorgroups
+  phase: Failed
+  reason: NoOperatorGroup
+lastTransitionTime: "2019-11-21T14:45:58Z"
+lastUpdateTime: "2019-11-21T14:46:40Z"
+message: csv in namespace with no operatorgroups
+phase: Failed
+reason: NoOperatorGroup
+```
+
+>Note: It is possible that the Status is missing, which suggests that the OLM operator is encountering an issue when processing the `CSV` in a very early stage. You should respond by reviewing the logs of the OLM operator.
+
+From the information within the `status.reason` and `status.message` fields, we now know that the `CSV` failed to install because it has been deployed in a namespace without an [OperatorGroup](https://operator-framework.github.io/olm-book/docs/operator-scoping).
```

Comment:
Same comment here as above. [Other docs](https://github.com/operator-framework/olm-book/blob/master/docs/how-do-i-install-my-operator-with-olm.md) deal with the installation flow. 
Maybe this document could benefit from pointed headers like 
"I created a subscription, but I do not see my operator being installed"
~ Steps to troubleshoot Subscription, and the CSV etc~
?

Reply:
This is a fair point - but those other docs specifically avoid running into issues. This document is meant to capture any issue a user may encounter when using OLM. 

I had originally went down the road of using headers as you suggested, but many of the problems with subscriptions required the same steps for debugging - and so I felt like there was limited value add.

Reply:
I'm trying to think about what my approach generally is when I'm reading a troubleshooting doc. Normally, with a stable product, I expect to run into issues that others have run into before. So my first thought always is, let me see if I can find the issue quickly, and I either look for links to section at the top of the document, or do a CTRL+F for keywords related to the issue I'm looking for. If I find my issue, I just jump to that section and only read that section and exit the document. 
Do you think there's any specific argument/objection against having a few initial steps repeatedly listed down?

Reply:
I'm happy with either approach - @kevinrizza can decide.

Reply:
My only hesitation is that our users have typically struggled to find out which component is failing.

Reply:
I tend to agree with Anik here. It seems like we are spending a lot of time in this doc setting up dominos so we can knock them down in an example. In reality, when someone is looking for this doc they are probably asking "I followed the steps to install my operator but the deployment isn't being created! Why?" I think it would be more reasonable in this doc to list a set of common issues under headers rather than have a toy example that someone who is already in a bad state is most likely frustrated and not going to spend the time to follow. They just want answers.

And if the goal of this documentation is partially so that when people run into issues we can point them to this, it's going to be really hard to give them a link partway into this doc that outlines their problem and troubleshooting steps.